html/mycakeapp/src/Controller/MembersController.php

```diff
@@ -160,4 +168,39 @@ public function changed()
     {
         $this->viewBuilder()->setLayout('frame-no-title');
     }
+    public function login()
+    {
+        $this->viewBuilder()->setLayout('frame-title');
+        $entity = $this->Members->newEntity();
+        if ($this->request->is('post')) {
+            // ログイン用のvalidationを適応
+            $entity = $this->Members->patchEntity($entity, $this->request->getData(), ['validate' => 'login']);
+            $member = $this->Auth->identify();
+            if (!empty($member)) { //認証成功
+                $this->Auth->setUser($member);
+                return $this->redirect($this->Auth->redirectUrl());
+            }
+            // ↓認証失敗時のメッセージ(email)はemail形式チェックのエラーメッセージと同じため
+            // email形式チェックのエラーメッセージがない時のみ認証失敗メッセージ(email)を追加する
+            $emailErrors = $entity->getError("email");
+            if (empty($emailErrors["email"])) {
+                $emailError = 'メールアドレスが間違っているようです';
```

Comment:
<提案>
エラー文を「メールアドレスかパスワードまたはその両方が間違っているようです」に変えるのはどうでしょうか？
もし片方はあってた場合、両方間違えてるみたいな言い方になっていて混乱すると思うので！
```suggestion
                $emailError = 'メールアドレスかパスワードまたはその両方が間違っているようです';
```

Reply:
バリデーションではじかれたのか、認証ではじかれたのか明確に分かっていいと思います！！
こちらも、修正いたします🙇‍♂️🙆‍♂️